modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4PipeliningIT.java

```diff
@@ -196,7 +196,10 @@ public List<Route> routes() {
 
                 @Override
                 protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
-                    return channel -> addListener(new RestToXContentListener<>(channel));
+                    return channel -> {
+                        request.close();
```

Comment:
this test handler might never return response and leak request at the end of the test

Reply:
:+1: could we add a comment here to that effect so we don't forget why we're closing the request here?